server/src/port/persistence/repository/user.repository.ts

```diff
@@ -1,12 +1,19 @@
-import { DeepPartial, EntityRepository, Repository, SaveOptions } from 'typeorm';
-import { Ghost } from '../../../domain/ghost/ghost.entity';
+import {EntityRepository, FindOneOptions, ObjectID, Repository} from 'typeorm';
+import {User} from '../../../domain/user/user.entity';
 
-@EntityRepository(Ghost)
-export class GhostRepository extends Repository<Ghost> implements IGhostRepository {
+@EntityRepository(User)
+export class UserRepository extends Repository<User> implements IUserRepository {
 
+    findById(id: number) {
+        return this.findOne({ id });
+    }
+
+    findByAddress(address: string) {
+        return this.findOne({address});
+    }
 }
 
-interface IGhostRepository {
-  findOne(id?: string): Promise<Ghost | undefined>;
-  save(entity: Ghost): Promise<Ghost>;
+interface IUserRepository {
```

Comment:
Please divide the interface `IUserRepository` and the implementation `UserRepository` in `user.repository.ts`.